src/api/v1beta1/properties.go

```diff
@@ -295,3 +295,15 @@ func splitArg(arg string) (key, value string) {
 	value = split[1]
 	return
 }
+
+func (dk *DynaKube) NeedsStatsD() bool {
+	if dk.FeatureEnableStatsDIngest() {
+		return true
+	}
+	for _, capability := range dk.Spec.ActiveGate.Capabilities {
+		if capability == StatsDIngestCapability.DisplayName {
+			return true
+		}
+	}
+	return false
```

Comment:
Checking both the feature flag and this capability feels redundant.
Remove the feature flag, check just for the capability instead. Marking this capability as preview similar to how it's handled for `metric-ingest` in `validation/preview.go` would be an option on how we could make it clearer that it shouldn't be used.

We have `IsActiveGateMode` for this check already, `NeedsStatsD` should be structured similarly to `KubernetesMonitoringMode`.

Reply:
Great, good to know. There was no `IsActiveGateMode()` at the time I wrote this but it's exactly what I need.

Ad feature flag vs capability: I'll go with the warning instead as it will be easier to use and test.